src/qililab/instruments/keithley/keithley_2600.py

```diff
@@ -55,12 +55,12 @@ def setup(  # type: ignore
         """Setup instrument."""
         if parameter == Parameter.MAX_CURRENT:
             self.max_current = float(value)
-            if hasattr(self, "device") and self.device is not None:
+            if Instrument.is_device_initialized(self):
```

Comment:
Does this syntax work? Why not `self.is_device_initialized()` ?

Reply:
Lmao, you are totally right, dunno why I did that! 

Changing it!